apis/python/src/tiledb/vector_search/ingestion.py

```diff
@@ -401,13 +401,13 @@ def read_source_metadata(
     ) -> Tuple[int, int, np.dtype]:
         if source_type == "TILEDB_ARRAY":
             schema = tiledb.ArraySchema.load(source_uri)
-            size = schema.domain.dim(1).domain[1] + 1
-            dimensions = schema.domain.dim(0).domain[1] + 1
+            size = np.int64(schema.domain.dim(1).domain[1]) + 1
+            dimensions = np.int64(schema.domain.dim(0).domain[1]) + 1
```

Comment:
We make this change b/c of changes to numpy data type promotion. Before numpy 2, the result of this code was:
```
[ingestion@read_source_metadata] schema.domain.dim(1).domain[1]: 2147483647 <class 'numpy.int32'>
[ingestion@read_source_metadata] size: 2147483648 <class 'numpy.int64'>
[ingestion@read_source_metadata] dimensions: 3 <class 'numpy.int64'>
```

With numpy 2 and the original code we instead were getting:
```
[ingestion@read_source_metadata] schema.domain.dim(1).domain[1]: 2147483647 <class 'numpy.int32'>
[ingestion@read_source_metadata] size: -2147483648 <class 'numpy.int32'>
[ingestion@read_source_metadata] dimensions: 3 <class 'numpy.int32'>
```

This is because in `size = schema.domain.dim(1).domain[1] + 1` the `+ 1` used to cause a cast to `int64` (which is required because `2147483647` is `int32` max).

As mentioned in https://numpy.org/devdocs/numpy_2_0_migration_guide.html#changes-to-numpy-data-type-promotion, numpy no longer casts automatically:

<img width="709" alt="Screenshot 2024-07-12 at 3 55 46 PM" src="https://github.com/user-attachments/assets/11f5f4aa-932f-4f46-8a6e-8a3588328a35">

So here we explicitly cast to `int64` so that we return the same value as we did before.